### DISCUSSION TITLE
Time complexity of compiling a regex

### DISCUSSION THREAD

**Discussion by septatrix** [rust-lang/regex#1158] [Q&A] [answered]
The documentation says that regex matching is performed in O(m*n), is there also some information about the complexity of compiling a regex (or is that even included in the above statement)? My use case would be to allow users to input arbitrary regexes which would then be compiled and used as a filter.

**Comment by BurntSushi** [comment, marked as answer]
Can you say why [this section in the docs](https://docs.rs/regex/latest/regex/#untrusted-input) didn't help you? It doesn't just narrowly address your time complexity question, but it also addresses your broader use case in a lot of detail. If you're accepting untrusted patterns from end users, I would recommend reading the entire section carefully.

**Reply by septatrix** [reply]
It did help, I just did not find it (as I was primarily looking at the docs for the `Regex` struct). Thanks!
